pip/operations/search.py

```diff
@@ -0,0 +1,60 @@
+# The guts of search
+
+from pip.download import PipXmlrpcTransport
+from pip._vendor.six.moves import xmlrpc_client
+from pip._vendor import pkg_resources
+
+
+def search(query, options, build_session_func):
+    index_url = options.index
+    with build_session_func(options) as session:
```

Comment:
The session building part could stay on the `comands` side. This would also allow to only have the `index_url` argument instead of the whole `options`.